src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
Maybe interesting. in 1.0.10 rollback / uncleanly leaving the contact manager would read a async error and then (via finally in __leave__) semi forceably close the connection (close all fds at least and wait for ssh to finish someway). The current code doesn‘t result in any kind of bailout and will just pile up all errors anyway and wait for a response to rollback.